services/motion/builtin/move_request.go

```diff
@@ -41,78 +44,127 @@ type validatedMotionConfiguration struct {
+	position, obstacle *replanner
+	// waypointIndex tracks the waypoint we are currently executing on
+	waypointIndex *atomic.Int32
 }
 
 // plan creates a plan using the currentInputs of the robot and the moveRequest's planRequest.
-func (mr *moveRequest) plan(ctx context.Context) ([][]referenceframe.Input, error) {
-	inputs, err := mr.kinematicBase.CurrentInputs(ctx)
+func (mr *moveRequest) Plan() (state.PlanResponse, error) {
+	inputs, err := mr.kinematicBase.CurrentInputs(mr.ctx)
 	if err != nil {
-		return nil, err
+		return state.PlanResponse{}, err
 	}
 	// TODO: this is really hacky and we should figure out a better place to store this information
 	if len(mr.kinematicBase.Kinematics().DoF()) == 2 {
 		inputs = inputs[:2]
 	}
 	mr.planRequest.StartConfiguration = map[string][]referenceframe.Input{mr.kinematicBase.Kinematics().Name(): inputs}
+
 	// TODO(RSDK-5634): this should pass in mr.seedplan and the appropriate replanCostFactor once this bug is found and fixed.
-	plan, err := motionplan.Replan(ctx, mr.planRequest, nil, 0)
+	plan, err := motionplan.Replan(mr.ctx, mr.planRequest, nil, 0)
 	if err != nil {
-		return nil, err
+		return state.PlanResponse{}, err
+	}
+
+	waypoints, err := plan.GetFrameSteps(mr.kinematicBase.Kinematics().Name())
+	if err != nil {
+		return state.PlanResponse{}, err
+	}
+
+	switch mr.requestType {
+	case requestTypeMoveOnMap:
+		return state.PlanResponse{
+			Waypoints:  waypoints,
+			Motionplan: plan,
+		}, nil
+	case requestTypeMoveOnGlobe:
+		return state.PlanResponse{
+			Waypoints:  waypoints,
+			Motionplan: plan,
+		}, nil
+	case requestTypeUnspecified:
+		fallthrough
+	default:
+		return state.PlanResponse{}, fmt.Errorf("invalid moveRequest.requestType: %d", mr.requestType)
 	}
-	mr.seedPlan = plan
-	return mr.seedPlan.GetFrameSteps(mr.kinematicBase.Kinematics().Name())
 }
 
 // execute attempts to follow a given Plan starting from the index percribed by waypointIndex.
 // Note that waypointIndex is an atomic int that is incremented in this function after each waypoint has been successfully reached.
-func (mr *moveRequest) execute(ctx context.Context, waypoints [][]referenceframe.Input, waypointIndex *atomic.Int32) moveResponse {
+func (mr *moveRequest) execute(ctx context.Context, waypoints state.Waypoints, waypointIndex *atomic.Int32) (state.ExecuteResponse, error) {
 	// Iterate through the list of waypoints and issue a command to move to each
 	for i := int(waypointIndex.Load()); i < len(waypoints); i++ {
 		select {
 		case <-ctx.Done():
-			return moveResponse{}
+			return state.ExecuteResponse{}, nil
 		default:
 			mr.planRequest.Logger.Info(waypoints[i])
 			if err := mr.kinematicBase.GoToInputs(ctx, waypoints[i]); err != nil {
 				// If there is an error on GoToInputs, stop the component if possible before returning the error
+				mr.logger.Debugf("calling kinematicBase.Stop due to %s\n", err)
 				if stopErr := mr.kinematicBase.Stop(ctx, nil); stopErr != nil {
```

Comment:
Are you happy about providing this context? I know there were some questions about which one we should be using.